Timeout error when deploying local Minimal Viable Dataspace

Hello,

We are experiencing a timeout error when deploying the local dataspace. We have tried on different computers and virtual machines, but the error persists. 
In two computers we have managed to deploy it, although it is not consistent. Most of the time you can't shut down and restart the data space, because this error appears.

We have tried it on a couple of linux operating systems (Ubuntu and Fedora). With docker 27.2.0 and java (jdk) in versions 11.0.24 and 21.0.2.
The computers we have tested all have intel i5 or i7 processors and between 16GB and 64GB of RAM.


```bash
[ERROR] >>> docker exec k3s-maven-plugin kubectl rollout status deployment provider-apisix-control-plane --namespace=provider --timeout=300s (timeout: PT5M10S)
[ERROR] <<< error: deployment "provider-apisix-control-plane" exceeded its progress deadline
[INFO] statefulset provider/provider-etcd ... ready
[INFO] deployment provider/provider-tm-forum-api-customer-bill-management ... ready
[INFO] statefulset provider/data-service-postgis ... ready
[INFO] deployment provider/provider-tm-forum-api-product-ordering-management ... ready
[INFO] deployment provider/odrl-pap ... ready
[INFO] deployment provider/provider-tm-forum-api-product-catalog ... ready
[INFO] deployment provider/provider-tm-forum-api-resource-function-activation ... ready
[ERROR] >>> docker exec k3s-maven-plugin kubectl rollout status deployment provider-apisix-data-plane --namespace=provider --timeout=300s (timeout: PT5M10S)
[ERROR] <<< error: deployment "provider-apisix-data-plane" exceeded its progress deadline
[INFO] deployment infra/traefik ... ready
[INFO] job provider/tmf-api-registration ... ready
[INFO] statefulset consumer/postgresql ... ready
[INFO] deployment provider/did-helper ... ready
[INFO] statefulset provider/postgresql ... ready
[INFO] deployment provider/provider-tm-forum-api-customer-management ... ready
[INFO] deployment provider/provider-tm-forum-api-product-inventory ... ready
[INFO] statefulset trust-anchor/trust-anchor-mysql ... ready
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for connector 0.0.1:
[INFO] 
[INFO] connector .......................................... FAILURE [01:34 min]
[INFO] it ................................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:34 min
[INFO] Finished at: 2024-09-11T13:12:33+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.kokuwa.maven:k3s-maven-plugin:1.3.0:apply (create-namespaces) on project connector: Failed to wait for resources, see previous log -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.kokuwa.maven:k3s-maven-plugin:1.3.0:apply (create-namespaces) on project connector: Failed to wait for resources, see previous log
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to wait for resources, see previous log
    at io.kokuwa.maven.k3s.mojo.ApplyMojo.execute (ApplyMojo.java:152)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/charts/data-space-connector/Chart.yaml
+++ b/charts/data-space-connector/Chart.yaml
@@ -32,7 +32,7 @@ dependencies:
     repository: https://fiware.github.io/helm-charts
   - name: apisix
     condition: apisix.enabled
-    version: 3.1.0
+    version: 3.5.1
     repository: oci://registry-1.docker.io/bitnamicharts
   # data-service
   - name: scorpio-broker-aaio

--- a/charts/data-space-connector/Chart.yaml
+++ b/charts/data-space-connector/Chart.yaml
@@ -2,7 +2,7 @@ apiVersion: v2
 name: data-space-connector
 description: Umbrella Chart for the FIWARE Data Space Connector, combining all essential parts to be used by a participant.
 type: application
-version: 7.3.1
+version: 7.3.2
 dependencies:
   - name: postgresql
     condition: postgresql.enabled

--- a/charts/data-space-connector/Chart.yaml
+++ b/charts/data-space-connector/Chart.yaml
@@ -2,7 +2,7 @@ apiVersion: v2
 name: data-space-connector
 description: Umbrella Chart for the FIWARE Data Space Connector, combining all essential parts to be used by a participant.
 type: application
-version: 7.3.2
+version: 7.3.3
 dependencies:
   - name: postgresql
     condition: postgresql.enabled

--- a/charts/data-space-connector/values.yaml
+++ b/charts/data-space-connector/values.yaml
@@ -173,6 +173,8 @@ apisix:
   controlPlane:
     # -- should it be enabled
     enabled: true
+    # -- resource preset to have sufficient memory
+    resourcesPreset: small
   # -- configuration in regard to the apisix ingressController
   ingressController:
     # -- should it be enabled
@@ -187,6 +189,8 @@ apisix:
       enabled: false
   # -- configuration in regard to the apisix dataplane
   dataPlane:
+    # -- resource preset to have sufficient memory
+    resourcesPreset: small
     # -- configuration for extra configmaps to be deployed
     extraConfig:
       deployment:
@@ -195,6 +199,12 @@ apisix:
           config_provider: yaml
       apisix:
         extra_lua_path: /extra/apisix/plugins/?.lua
+    # -- configuration in regard to the apisix dashboard
+    dashboard:
+      # -- should it be enabled
+      enabled: true
+      # -- resource preset to have sufficient memory
+      resourcesPreset: small
     # -- extra volumes
     # we need `routes` to declaratively configure the routes
     # and the config for the opa sidecar

--- a/k3s/provider.yaml
+++ b/k3s/provider.yaml
@@ -4,8 +4,6 @@ keycloak:
 apisix:
   image:
     debug: true
-  dashboard:
-    enabled: false
   dataPlane:
     ingress:
       enabled: true

